XMLtoMD/GuidelineXmlToMD/MarkdownOut/MdText.cs

```diff
@@ -0,0 +1,225 @@
+            }
+            return prefix + text;
+        }
+
+        public static string RemoveInvalidMDLinkCharacters(object text)
+        {
+            return Regex.Replace(text.ToString(), "[^A-Za-z]", "");
+        }
+
+        /// <summary>
+        /// Styles then formats the provided text using the specified <see cref="MdStyle"/> and
+        /// <see cref="MdFormat"/>.
+        /// </summary>
+        /// <param name="text">The text to format and style.</param>
+        /// <param name="style">The Markdown style to apply.</param>
+        /// <param name="format">The Markdown format to apply.</param>
+        /// <returns>The formatted and styled string.</returns>
+        public static string StyleAndFormat(object text, MdStyle style, MdFormat format) {
+            return Format(Style(text, style), format);
+        }
+
+        /// <summary>
+        /// Applies an indent to the provided text.
+        /// </summary>
+        /// <param name="text">The text to indent.</param>
+        /// <param name="indent">
+        /// The indent length to apply (0 adds no indent, 1 adds a single indentation to create a
+        /// sublist, etc). Negative values are ignored.
+        /// </param>
+        /// <returns>The indented string.</returns>
+        public static string Indent(object text, int indent) {
+            string indentPrefix = String.Empty;
+            while (indent > 0) {
+                indentPrefix += ListItemIndent;
+                indent--;
+            }
+            return indentPrefix + text;
+        }
+
+        /// <summary>
+        /// Cleanses the provided text by replacing all tab characters with the <see cref="Tab"/>
+        /// string and ensuring that all newlines are the Windows <c>\r\n</c>.
+        /// </summary>
+        /// <param name="text">The text to cleanse.</param>
+        /// <param name="useMdLineBreaks">
+        /// If true, all newlines will be replaced by Markdown line breaks instead of just Windows
+        /// newlines. This is useful when cleansing text formatted as a list item as it maintains
+        /// the desired line breaks in the list item's text.
+        /// </param>
+        /// <returns>The cleansed text with consistent newlines and no tab characters.</returns>
+        public static string Cleanse(string text, bool useMdLineBreaks = false) {
+            //first, replace all tab characters with the Tab string
+            text = text.Replace("\t", Tab);
+            /*
+             * Next, replace all Windows newlines with the Unix newline; this is done to make sure
+             * the following code doesn't replace instances of "\r\n" with "\r\n\n". All of this is
+             * done to make sure the raw text displays correctly on Windows machines (programs such
+             * as Visual Studio or Notepad do not recognize "\n" as a newline, just "\r\n").
+             */
+            text = text.Replace("\r\n", "\n").Replace("\r", "\n");
```

Comment:
Is this needed? When checking out files with git it will attempt to handle line endings for you based on your operating system.